servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -2349,6 +2392,42 @@ Policy updateTemplatePolicy(Policy policy, String domainName, String policyName,
         return templatePolicy;
     }
     
+    ServiceIdentity updateTemplateServiceIdentity(ServiceIdentity serviceIdentity, String domainName, String serviceIdentityName, List<TemplateParam> params) {
+        
+        String templateServiceName = serviceIdentityName;
+        if (params != null) {
+            for (TemplateParam param : params) {
+                final String paramKey = "_" + param.getName() + "_";
+                System.out.println("param key &&&&&&&&&& " + paramKey + " &&&&&&");
+                templateServiceName = templateServiceName.replace(paramKey, param.getValue());
+            }
+        }
+        
+        ServiceIdentity templateServiceIdentity = new ServiceIdentity().setName(ZMSUtils.serviceResourceName(domainName, serviceIdentityName));
```

Comment:
Here we should be using updated service name: templateServiceName and not serviceIdentityName